programs/github

```diff
@@ -21,14 +21,47 @@ function progressbar(f)
 	return left .. right
 end
 
-if action == 'clone' then
-	-- parse input
-	local repo, treeName, dest = select(2, ...)
+-- get flag arguments
+local acceptedFlags = {
+	['-b'] = 'branch',
+	['--branch'] = 'branch',
+	['-t'] = 'tag',
+	['--tag'] = 'tag'
+}
 
-	if not dest then
-		dest = treeName
-		treeName = 'master'
+local function findFlag(arg)
+	if arg.sub(1,1) == '-' then
+		if not acceptedFlags[arg] then
+			return printError(('Unknown flag argument %s'):format(arg))
```

Comment:
Right, but it only exits this function. The script will continue and run, won't it, treating this as the dest/repo?